ios/engine/KMEI/KeymanEngine/Classes/Manager.swift

```diff
@@ -0,0 +1,2970 @@
+    let keyboardURL = URL(string: kbBaseUri! + fileName)!
+    var keyboardFontURLs: [URL] = []
+    // TODO: Refactor duplicate with kbOskFont
+    if let kbFont = kbFont, !kbFont.isEmpty {
+      // Font filename is deprecated
+      let fontSource = kbFont[Key.fontSource] ?? kbFont[Key.fontFilename]
+      if let fontSource = fontSource as? [String] {
+        keyboardFontURLs = fontSource.flatMap { source in
+          if source.hasFontExtension {
+            return URL(string: fontBaseUri! + source)
+          } else {
+            return nil
+          }
+        }
+      } else if let fontSource = fontSource as? String {
+        if fontSource.hasFontExtension {
+          keyboardFontURLs = [URL(string: fontBaseUri! + fontSource)!]
+        } else {
+          keyboardFontURLs = []
+        }
+      } else {
+        let message = "Unexpected error: \(String(describing: fontSource)) is not a valid type to be processed."
+        let error = NSError(domain: "Keyman", code: 0, userInfo: [NSLocalizedDescriptionKey: message])
+        downloadFailed(forKeyboard: kbInfo, error: error)
+        return
+      }
+    }
+    if let kbOskFont = kbOskFont, !kbOskFont.isEmpty {
+      // Font filename is deprecated
+      let fontSource = kbOskFont[Key.fontSource] ?? kbOskFont[Key.fontFilename]
+      if let fontSource = fontSource as? [String] {
+        keyboardFontURLs.append(contentsOf: fontSource.flatMap { source in
+          if source.hasFontExtension {
+            let url = URL(string: fontBaseUri! + source)!
+            if !keyboardFontURLs.contains(url) {
+              return url
+            }
+          }
+          return nil
+        })
+      } else if let fontSource = fontSource as? String {
+        if fontSource.hasFontExtension {
+          let url = URL(string: fontBaseUri! + fontSource)!
+          if !keyboardFontURLs.contains(url) {
+            keyboardFontURLs.append(url)
+          }
+        }
+      } else {
+        let message = "Unexpected error: \(String(describing: fontSource)) is not a valid type to be processed."
+        let error = NSError(domain: "Keyman", code: 0, userInfo: [NSLocalizedDescriptionKey: message])
+        downloadFailed(forKeyboard: kbInfo, error: error)
+        return
+      }
+    }
+
+    let isUpdate = latestKeyboardFileVersion(withID: kbID) != nil
+
+    downloadQueue = HTTPDownloader.init(self)
+    let commonUserData = [
+      Key.keyboardInfo: kbInfo[Key.keyboardInfo]!,
+      Key.update: (isUpdate ? 1 : 0)
+    ]
+    downloadQueue!.userInfo = commonUserData
+
+    var request = HTTPDownloadRequest(url: keyboardURL, userInfo: commonUserData)
+    request.destinationFile = keyboardLocalPath.path
+    request.tag = 0
+    downloadQueue!.addRequest(request)
+    for (i, url) in keyboardFontURLs.enumerated() {
+      let fontPath = self.fontPath(forFilename: url.lastPathComponent)!
+      // FIXME: Downloading from local path to local path?
```

Comment:
- [ ] Reminder:  make an issue for this if not fixing it within this base PR.

Reply:
Opened #398